#### 08/01/2019  
##### The change   
After a meeting with Marcela, she suggested us to try a different assembler called [Discovar *de novo*](https://software.broadinstitute.org/software/discovar/blog/). According to her and some other papers like [this](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-2531-7), Discovar seems to deal well with simpler sequencing input, especifically paired end Illumina reads. Since this is exactly our case for the assembly of male and female genomes, it looks an obvious choice to give a shot to Discovar.  

##### Installing Discovar  
1. First, we will install the [miniconda](https://conda.io/miniconda.html) package manager:  
2. Then, install [Discovar *de novo*]() (beware not to install Discovar, which makes reference-guided assemblies)  
    ```console  
    rpgisi21@DKT38375:~$ conda install -c bioconda discovardenovo  
    ```  
3. You can now run Discovar by typing:  
    ```console  
    rpgisi21@DKT38375:~$ DiscovarDeNovo  
    ```  

#### 09/01/2019 
##### Status  
Installed Miniconda and DiscovarDeNovo at the lf2-genome-assembly-discovar machine, but still missing transferring the raw data and installing Gcloud SDK
