#### 14/01/2019  
##### Assembly-stats  
In order to collect statistics regarding the assembly done by DISCOVAR, we used the program assembly-stats (version 1.0.1):  

```console  
joao_nunes@lfX-genome-assembly-discovar:~/lfX-assembly-discovar/lfX-genome-assembly-discovar/a.final$ assembly-stats a.lines.fasta  
```  
Which gave the following output:  

##### LF2  
sum = 2332356097, n = 2055920, ave = 1134.46, largest = 211093  
N50 = 6282, n = 76305  
N60 = 3850, n = 123873  
N70 = 1890, n = 209682  
N80 = 646, n = 426197  
N90 = 279, n = 993795  
N100 = 26, n = 2055920  
N_count = 520500  
Gaps = 5205    

##### LF6  
sum = 2276378180, n = 1902983, ave = 1196.22, largest = 241846  
N50 = 6315, n = 74933  
N60 = 3944, n = 120713  
N70 = 2051, n = 200150  
N80 = 752, n = 385112  
N90 = 305, n = 890734  
N100 = 30, n = 1902983  
N_count = 472800  
Gaps = 4728     

##### Published genome 
Those stats are much worse than the assembly done by Marcela (the first published version of the *L. fortunei* genome):  

sum = 1673125894, n = 20548, ave = 81425.24, largest = 2720304  
N50 = 312020, n = 1481  
N60 = 228817, n = 2107  
N70 = 141275, n = 3022  
N80 = 64814, n = 4766  
N90 = 25090, n = 9164   
N100 = 558, n = 20548  
N_count = 3851050  
Gaps = 40616  

#### 15/01/2019  
##### BUSCO  
In order to estimate how well the assembly has kept the coding regions of the genome, we run the program BUSCO:  
##### LF2   
```console  
joao_nunes@trinity-2:~/busco-discovar-assembly$ run_busco -i discovar-assemblies/lf2.a.lines.fasta -o lf2-discovar-busco -l /home/ti_admin/busco/data/metazoa_odb9 -m geno -c 32; sudo poweroff  
```  

Which gave us the following basic statistics:  
  
C:49.8%[S:39.6%,D:10.2%],F:33.9%,M:16.3%,n:978

487     Complete BUSCOs (C)  
387     Complete and single-copy BUSCOs (S)  
100     Complete and duplicated BUSCOs (D)  
332     Fragmented BUSCOs (F)  
159     Missing BUSCOs (M)  
978     Total BUSCO groups searched  

##### Published genome  
We then run the BUSCO analysis for the published *L. fortunei* genome in order to compare them:  
```console  
joao_nunes@trinity-2:~/busco-discovar-assembly$ run_busco -i LF.genome.fasta -o LF.genome.fasta-discovar-busco -l /home/ti_admin/busco/data/metazoa_odb9 -m geno -c 14  
```  
#### 18/01/2019 
All the files regarding BUSCO analysis have been transferred to the instance *assembly-quality-analysis*, which will be used from now on. BUSCO analysis has finished for the published genome (LF.genome.fasta) and it started running for the male (LF6) genome.  

#### 21/01/2019 
The following image summarizes the BUSCO analysis for the published (LF), the male (LF6) and the female (LF2) genome:  

<p align="left">
  <img width="600" height="500" src="https://github.com/biobureaubiotech/goldenMusselGender/blob/master/images/busco_LF-lf2-lf6.png">
</p> 

Taking into account both Assembly-stats and BUSCO analysis, we can tell that the assembly obtained using Discovar for both the male and female was significantly worse than our first published genome. Since the published genome was assembled using not only paired end but also mate pair and PacBio reads, that should not be a surprise.  
