#### 24-01-19  
The hybrid assembly strategy has taken too long to complete, reason why we decided to stop it. However, we don't want to waste the information generated by it. One interesting thing to explore is the fact that, even though more than 90% of the reads have mapped to the reference genome, this only represented around 70% of the genome. In other words, 30% of the refence genome didn't have any reads mapped to it. What exactly are those unmapped regions? Let's investigate it!  

##### Retrieve the FASTA sequences of the unmapped regions from the reference genome 
Once we have the BED file, we can generate a FASTA file with the respective regions from the reference genome. This can be achieved using the faidx command from samtools, as suggested in [this](https://www.biostars.org/p/10469/) thread  
 
#### 29-01-19  
##### Retrieving the BAM mapping files: LF6  
The mapping files (BAM) were downloaded from the lf6-genome-assembly-2 instance and merged using the *samtools merge* command. After merging, the concatenated BAM file was sorted using *samtools sort*:  
```console  
bioma@bioma-XPS-8300:~/lfortunei_gender/lf6/map-reference$ samtools sort Lfortunei_LF6_all.samtoolsV19-merge.bam Lfortunei_LF6_all.samtoolsV19-merge.sorted.bam -@ 6  
```  
After making sure that the final sorted file had the same number of reads as the sum of the input files, we proceeded to the next step.  

#### 01-02-19  
##### Retrieving the coordinates of the unmapped regions from the reference genome  
In order to retrive the regions from the reference genome where no reads mapped to, we first need to get the coordinates of those reads from the BAM file. This was achieved by the command:  

```console 
bioma@bioma-XPS-8300:~/lfortunei_gender/lf6/map-reference$ bedtools genomecov -ibam Lfortunei_LF6_all.samtoolsV19-merge.sorted.bam -bga | awk '$4==0' > Lfortunei_LF6_unmapped-regions.bed
```  

which we got from [this](https://www.biostars.org/p/189781/) thread  

##### Getting unmapped genes  
Besides retrieving the coordinates of the unmapped regions of the reference genome (and after it their FASTA sequences to evaluate their content), another interesting question to answer is: are there any genes that don't have any reads mapped to?  

To answer this question, a possibility is to use the [HTSeq tool](https://www.huber.embl.de/users/anders/HTSeq/doc/count.html)  

#### 01-04-19 
##### Getting unmapped genes with SAMTOOLS  
HTSeq seemed to be a complex tool to get used with at first glance. I found out that I could use a one-line command using samtools to achieve the same purpose of counting the number of mapping to each gene:  

```console  
$ bedtools multicov -bams yourBAMfile.bam -bed yourGenes.bed  
```  

We already had a bam file (*Lfortunei_LF6_all.samtoolsV19-merge.sorted.bam*), so we were just missing a bed file containing all L.fortunei's genes. To get that file, we proceeded with the command:  

```console  
rpgisi21@DKT38375:~/Documentos/lfortunei_gender/26_genes$ grep "gene" lfortunei_genome.gff | grep -v "#" | awk '{print($1" "$4" "$5)}' > all-genes.bed  
```  

We then retrieved the number of mapping to each gene using the command:  
```console  
rpgisi21@DKT38375:~/Documentos/lfortunei_gender/26_genes$ bedtools multicov -bams Lfortunei_LF6_all.samtoolsV19-merge.sorted.bam -bed all-genes.bed > all-genes-with-count.bed  
```  

And to check how many genes didn't have any mapping we ran:  
```console  
bioma@bioma-XPS-8300:~/lfortunei_gender/lf6/map-reference$ awk 'BEGIN{c=0}{if($4==0){c+=1}}END{print(c)}' all-genes-with-count.bed  
```  
Which gave as result **3305 genes**. 
  
### To Do  
* Double check the *all-genes.bed* file, because it has 56130, as opposed to the 60717 genes described at Marcela's [paper](https://peerj.com/preprints/2995/)  

 


