#### 02/20/2019  
#### Switching to another gap-filling algorithm: Sealer  
Gap2Seq was going to take too long to complete according to the execution status. Taking too long to gap-fill large genomes is actually a common problem with gap-filling algorithms, that are usually more suitable for bacterial genomes. A promising alternative algorithm is **Sealer**, part of the Abyss assembly pipeline. According to Sealer description paper:  

> Low sequence coverage, repetitive elements and short read length make de novo genome assembly difficult, often resulting in sequence and/or fragment “gaps” – uncharacterized nucleotide (N) stretches of unknown or estimated lengths. Some of these gaps can be closed by re-processing latent information in the raw reads. Even though there are several tools for closing gaps, they do not easily scale up to processing billion base pair genomes. ([Paulino et al., 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4515008/#MOESM4))

Also according to its paper, Sealer was capable of gap-filling the human genome in 29 hours using a 12 cores, 48G RAM machine. In order to test Sealer for gap-filling our LF2 dataset, we have run the command: 

```console  
joao_nunes@lf2-genome-assembly-discovar:~/abyss-sealer$ abyss-sealer -S /home/joao_nunes/align_graph/lf2/a.lines.fasta -k 80 -o a.lines.gap-filled.fasta -j 16 -b 40G /home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R1_001.fasta /home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R2_001.fasta  
```  

Where k-mer=80bp size was chosen because it has proven to be the optimal size for the human genome, as well as a 40G for the bloom filter. 

The command above has taken only 2h30min to finish. However, it only used one k-mer size to do the gap-filling, instead of the recommended range that goes from average size of the reads down to 40 bp. We then re-run Sealer using as input the output generated from the first round of Sealer (with k-mer=80bp) and using k-mer=40bp this time (i.e., running Sealer iteratively with different k-mer sizes):  

```console  
joao_nunes@lf2-genome-assembly-discovar:~/abyss-sealer-kmer40$ abyss-sealer -S /home/joao_nunes/abyss-sealer-kmer80/a.lines.gap-filled.fasta_scaffold.fa -k 40 -o a.lines.gap-filled.fasta -j 16 -b 40G /home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R1_001.fasta /home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R2_001.fasta  
```  

### To Do
#### How to run Sealer with larger k-mer lengths?  
When we tried to run Sealer with 250, 200 or 150 bp k-mer lengths, we got the following error:    
> abyss-sealer: ../Common/Kmer.h:49: static void Kmer::setLength(unsigned int): Assertion `length <= MAX_KMER' failed.  

The solution to this error was to re-configure and recompile Abyss:  
https://www.biostars.org/p/194314/

The compilation was done through: https://github.com/bcgsc/abyss/tree/sealer-release#compiling-abyss-from-github  

#### 02/21/2019  
We have iterated over Sealer runs for k-mer sizes of (in order of execution): 80bp, 40bp, 250bp and 128bp. The first iteration (k-mer=80bp) has closed 475 gaps (9,13%), the second one (k-mer=40bp) has closed 51 gaps (1,08%) and the last two iterations (k-mers=250bp and 128bp) haven't closed any gaps (probably all saturated by the first two rounds of iteration.  

#### Analyzing gap-filled genome  
In order to assess the quality of the Discovar genome gap-filled with Sealer, we have run *assembly-stats*. The main statistics for the final gap-filled file was:  
> stats for a.lines.gap-filled.fasta_scaffold.fa  
sum = 2332337804, n = 2055920, ave = 1134.45, largest = 211128  
N50 = 6282, n = 76306  
N60 = 3849, n = 123875  
N70 = 1890, n = 209685  
N80 = 646, n = 426203  
N90 = 279, n = 993801  
N100 = 26, n = 2055920  
N_count = 467900  
Gaps = 4679  

In comparison to the original [Discovar LF-2 genome](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/7.Analyzing_Discovar's_Assembly.md#lf2), the contiguity of the genome (number of scaffolds(n), N50,60..) and its length weren't changed. However, the number of gaps was reduced from 5205 to 4679, i.e., 10,11% of gaps closed. 

#### 03/27/2019  
#### Running Sealer for LF6 genome  
After filling gaps for the LF2 Discovar genome, we proceeded to gap fill the LF6 Discovar genome. Since the runs with k-mer=250bp and k-mer=128bp didn't close any gaps for the LF2 genome, we didn't run those this time (let's save computation resources!).  

The command line to run *Abyss-Sealer* for the LF6 genome was:  
```console  

```
