#### 02/22/2019  
#### Align current reference and Discovar assemblies using NUCMER4
In paralel with running AlignGraph to improve Discovar's LF-2 assembly, we are running a genome-scale alignment of our current reference genome (assembled by Marcela) and the Discovar assembly (after Sealer gap-filling step).  
The program used to run the alignment was *NUCMER4*:  

```console  
joao_nunes@lf2-genome-assembly-discovar:~/nucmer-Marcela-vs-Discovar$ nucmer4 --threads=8 ../align_graph/LF.genome.fasta ../abyss-sealer-kmer128/a.lines.gap-filled.fasta_scaffold.fa 2>&1 | tee log_nucmer4_Marcela_vs_Discovar.txt  
```  

#### Where is the process running?  
On lf2-genome-assembly-discovar, at the same time as the AlignGraph process.

#### Estimated run time  
According to [MUMMER4 paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5802927/), the time taken to run an alignment of human and chipanzee genomes was 207 minutes. However, they have used 32 threads, as opposed to our 8 threads. On the other side, the human and chimpanzee genomes are smaller than *limnoperna fortunei*. Let's be pessimist and estimate 12 hours.  

#### 03/28/2019  
#### Align LF2, LF6 and Reference genomes  
Now we want to compare the two genomes assembled by Discovar (LF2 and LF6) with our current reference genome (LF.genome.fasta). Before doing that, we have renamed our final version (i.e, after [gap-filling step](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/10.Gap-filling-Discovar-scaffolds.md) with Abyss-Sealer) LF2 and LF6 genomes to *lf2_scaffolds.fasta* and *lf6_scaffolds.fasta*, respectively. Then we ran the alignment using the program NUCMER, version 4.0.0beta2:  

LF2 vs Reference  
```console  
joao_nunes@improve-reference-genome-2:~/nucmer4_alignment$ nucmer --threads=16 ../genomes/LF.genome.fasta ../genomes/lf2_scaffolds.fasta 2>&1 | tee log_nucmer4_Marcela_vs_lf2-Discovar.txt; sudo poweroff
```  

LF6 vs Reference  
```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment$ nucmer --threads=16 ../genomes/LF.genome.fasta ../genomes/lf6_scaffolds.fasta 2>&1 | tee log_nucmer4_Marcela_vs_lf6-Discovar.txt; sudo poweroff
```  

LF6 vs LF2  
```console  
joao_nunes@improve-reference-genome-3:~/nucmer4_alignment$ nucmer --threads=16 ../genomes/lf2_scaffolds.fasta ../genomes/lf6_scaffolds.fasta 2>&1 | tee log_nucmer4_lf2-Discovar_vs_lf6-Discovar.txt; sudo poweroff
```

#### 04/02/2019  
#### Analyzing the alignments' output  
After running the 3 alignments above using nucmer, we proceeded to analyse the output of the alignment (i.e., the *.delta* file). In order to do that, we have first ran the script *mummerplot* from the *MUMMER* package to generate a 2d plot of the alignment. The first alignment tested was *LF6 vs Reference*:  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment$ mummerplot out.delta --png
```  

Which generated the following *png* file:  
![LF6_vs_Reference_nucmer](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/images/out.png)

According to [this thread](https://sourceforge.net/p/mummer/mailman/mummer-help/thread/CAFpLa4KSyWNZFWa4cMFgr6eQRsCnAiBovBBbRzA9SzmKTnqkqA%40mail.gmail.com/), this weird pattern of vertical lines could be caused by repeat regions. To try to clean up the plot, we have run the *delta-filter* command to keep only the 1-1 map regions:  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment$ delta-filter -1 out.delta > out.m.delta
```  
  

