#### 02/22/2019  
#### Align current reference and Discovar assemblies using NUCMER4
In paralel with running AlignGraph to improve Discovar's LF-2 assembly, we are running a genome-scale alignment of our current reference genome (assembled by Marcela) and the Discovar assembly (after Sealer gap-filling step).  
The program used to run the alignment was *NUCMER4*:  

```console  
joao_nunes@lf2-genome-assembly-discovar:~/nucmer-Marcela-vs-Discovar$ nucmer4 --threads=8 ../align_graph/LF.genome.fasta ../abyss-sealer-kmer128/a.lines.gap-filled.fasta_scaffold.fa 2>&1 | tee log_nucmer4_Marcela_vs_Discovar.txt  
```  

#### Where is the process running?  
On lf2-genome-assembly-discovar, at the same time as the AlignGraph process.

#### Estimated run time  
According to [MUMMER4 paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5802927/), the time taken to run an alignment of human and chipanzee genomes was 207 minutes. However, they have used 32 threads, as opposed to our 8 threads. On the other side, the human and chimpanzee genomes are smaller than *limnoperna fortunei*. Let's be pessimist and estimate 12 hours.  

#### 03/28/2019  
#### Align LF2, LF6 and Reference genomes  
Now we want to compare the two genomes assembled by Discovar (LF2 and LF6) with our current reference genome (LF.genome.fasta). Before doing that, we have renamed our final version (i.e, after [gap-filling step](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/10.Gap-filling-Discovar-scaffolds.md) with Abyss-Sealer) LF2 and LF6 genomes to *lf2_scaffolds.fasta* and *lf6_scaffolds.fasta*, respectively. Then we ran the alignment using the program NUCMER, version 4.0.0beta2:  

LF2 vs Reference  
```console  
joao_nunes@improve-reference-genome-2:~/nucmer4_alignment$ nucmer --threads=16 ../genomes/LF.genome.fasta ../genomes/lf2_scaffolds.fasta 2>&1 | tee log_nucmer4_Marcela_vs_lf2-Discovar.txt; sudo poweroff
```  

LF6 vs Reference  
```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment$ nucmer --threads=16 ../genomes/LF.genome.fasta ../genomes/lf6_scaffolds.fasta 2>&1 | tee log_nucmer4_Marcela_vs_lf6-Discovar.txt; sudo poweroff
```  

LF6 vs LF2  
```console  
joao_nunes@improve-reference-genome-3:~/nucmer4_alignment$ nucmer --threads=16 ../genomes/lf2_scaffolds.fasta ../genomes/lf6_scaffolds.fasta 2>&1 | tee log_nucmer4_lf2-Discovar_vs_lf6-Discovar.txt; sudo poweroff
```

#### 04/02/2019  
#### Analyzing the alignments' output  
After running the 3 alignments above using nucmer, we proceeded to analyse the output of the alignment (i.e., the *.delta* file). In order to do that, we have first ran the script *mummerplot* from the *MUMMER* package to generate a 2d plot of the alignment. The first alignment tested was *LF6 vs Reference*:  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment$ mummerplot out.delta --png
```  

Which generated the following *png* file:  
![LF6_vs_Reference_nucmer](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/images/out.png)

According to [this thread](https://sourceforge.net/p/mummer/mailman/mummer-help/thread/CAFpLa4KSyWNZFWa4cMFgr6eQRsCnAiBovBBbRzA9SzmKTnqkqA%40mail.gmail.com/), this weird pattern of vertical lines could be caused by repeat regions. To try to clean up the plot, we have run the *delta-filter* command to keep only the 1-1 map regions:  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment$ delta-filter -1 out.delta > out.m.delta
```  

However, not even this delta file has produced an useful image:  

![LF6_vs_Reference_nucmer](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/images/out.m.png)

Paralelly we have generated a simplified *delta* file, containing only the mappings to the scaffold itr6_-10002. To generate that delta file, we have developed a python script called [get-specific-scaffolds.py](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/scripts/get-specific-scaffolds.py). That script was run as:  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment$ python3 get-specific-scaffolds.py --input_file out.delta --output_file out.filtered-itr6-10002.delta --scaffold_id itr6_10002_
```
We have then built a mummerplot for that simplified file:  

```console
joao_nunes@improve-reference-genome:~/nucmer4_alignment$ mummerplot out.filtered-itr6-10002.delta --png
```  

However, this simplified version of the delta file hasn't been able to generate a good image neither:  

![LF6_vs_Reference_nucmer](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/images/out_simplified.png)

#### 04/09/2019  
#### Simplifying the genomes to create a interpretable image  
Our hypothesis is that the weird images produced so far are a result from the huge number of scaffolds aligned. In order to test this hypothesis, we decided to re-run the nucmer alignment with simplified versions of both the reference and LF6 Discovar genome.  

To produce simplified versions of the genomes, we have extracted the IDs from the first 13 scaffolds of the reference genome that had any *LF6 Discovar* scaffolds mapped to:  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment$ grep ">" out.delta.show-tiling | head -13 | awk '{print($1)}' > 13_reference_scaffolds.txt  
```  

and then extracted all the IDs from the scaffolds of *LF6 Discovar* genome that mapped to the 13 reference scaffolds using the *in-house* script [get_discovar_scaffolds.py](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/scripts/get_discovar_scaffolds.py):  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment/13_reference_scaffolds_alignment$ python3 get_discovar_scaffolds.py > lf6_discovar_simplified_scaffolds.txt  
``` 

Then we have extracted the simplified reference and *LF6 Discovar* genomes using the IDs as a filter and running the *in-house* scripts [process_reference_genome.sh]() and [process_lf6_discovar_genome.sh]():  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment/13_reference_scaffolds_alignment$ ./process_reference_genome.sh > 13_reference_scaffolds.fasta
``` 
```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment/13_reference_scaffolds_alignment$ ./process_lf6_discovar_genome.sh > lf6_discovar_simplified.fasta
``` 

>important note: before running *process_reference_genome.sh* and *process_lf6_discovar_genome.sh*, you need to index the original genomes that will be simplified (in this case, LF.genome.fasta and *lf6_scaffolds.fasta* using the *samtools faidx* command. 

#### Re-running nucmer with the simplified genomes
Now that we had the simplified genomes, we could run the *nucmer* alignment again using them:  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment/13_reference_scaffolds_alignment$ nucmer --threads=8 13_reference_scaffolds.fasta lf6_discovar_simplified.fasta 2>&1 | tee log_nucmer4_simplified_genomes_reference_vs_lf6-Discovar.txt; sudo poweroff
```

#### Analysing the alignment plot for the simplified genomes  
Finally, we have run *mummerplot* to generate a 2-D plot of the alignment above:  

```console  
joao_nunes@improve-reference-genome:~/nucmer4_alignment/13_reference_scaffolds_alignment$ mummerplot out.delta --png --filter  
```  

Which generated a much better image:  
![Reference_vs_LF6-Discovar_Simplified_genomes](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/images/filtered_out.png)
