#### 08/23/2018
So far, we have been able to successfully run the pipeline until the end of the step 2. In order to do it, we've changed the original script, commenting out the lines that referred to mate-pair reads (since we only have paired-end). 

By the end of the step 2, we've got the blocks and the superblocks built. Now we are going to run the _de novo_ assembly step, where reads that aligned to the superblocks will be assembled together. 

In order to run this assembly step we need to install the _de novo_ assembler AllPathsLG into the instance. Problem: can't compile it on Ubuntu. I have already tried to do it on a local machine and on our virtual machine at GCloud. However, the compilation shows many errors regarding ambiguity and redundance (error file is at the local computer). I have already tried to run the compilation using both gcc 7.3 and gcc 4.8, but still struggling with the same problem. 

If I still cannot make it work until Friday, I'm afraid we will need to try a different assembler, since Heidi Lischer's pipeline allow other assemblers to be used.    

##### Instructions to access terminal as superuser at the local computer at SENAI

1. To login as admin (root), run:
    ```console 
    rpgisi21@instance:~$ su administrador
    ``` 
2. To run commands at terminal as admin (root), run:
    ```console 
    administrador@KT38375:~$ sudo -i
    ``` 
3. Once the terminal look like it:  
    ```console 
    root@DKT38375:~#
    ``` 
    You now you are already running commands as a superuser (root/admin)

#### 08/24/2018
Fixing compilation of Allpaths-lg.  
Procedure: 
1. To redirect the compilation output to the file original_output.txt, run 
```console
root@DKT38375:~# make CMAKE_CXX_FLAGS="-fpermissive" &> original_output.txt  
```

2. To select only the error lines to the file errors.txt, run 
```console
root@DKT38375:~# cat original_output.txt | grep error > errors.txt   
```
 
3. Evaluate the file errors.txt and fix the problems 

PS: replacing align by Align has been solving our problems. With the Vim editor, we can accomplish it by typing the command:  
```console
:%s/\<align\>/Align/gc
``` 
#### 08/27/2018
Today I kept fixing the AllPaths-LG's source code. After working on it from 8h30 am to 12pm, I stopped at the error:
```console
paths/KmerPathInterval.cc:84:10: error: ‘ulonglong big_tagged_rpint::<unnamed union>::data1_’ is private within this context  
```   
#### 08/28/2018
Gave up on using AllPaths-LG because the compilation keeps showing multiple errors even though I have been debugging it for 2 days. I think that much effort would only be reasonable if we were sure that AllPahts was going to be more efficient than the alternative assemblers. Even though this was the case as reported by Heidi Lischer, it is well known that different assemblers may considerably change its behaviour depending on the input dataset (then H. Lischer's results should not take as to an absolute conclusion regarding the assemblers). 

Since SOAPdenovo2 already provided pre-compiled binary files, we have chosen it to proceeed with H. Lischer's pipeline. Since differences between the pipelines only start to appear from the Assembly step on, we can use files generated over the running of H. Lishcer AllPaths' pipeline. 

##### How to highlight differences between the AllPaths and SOAPdenovo pipelines? 
A possibility is to run the command: 

```console
root@DKT38375:~# diff -y refGuidedDeNovoAssembly_SOAP.sh refGuidedDeNovoAssembly_ALLPATHS.sh
```  
This command will place the files side to side and highlight lines that are different with a "|" symbol, as well as inserted or deleted lines with a ">" symbol.

#### 08/30/2018
Today I adjusted all variables from the refGuidedDeNovoAssembly_SOAP.sh file, creating a new file called refGuidedDeNovoAssembly_SOAP_withSettings.sh file. 
 
As the current settings of our lf2-genome-assembly machine were too weak to run a _de novo_ assembly (a n1-standard-4 machine with 4 vCPUs, 15 GB memory), we had to upgrade it. Since the _de novo_ assembler to be used is SOAPdenovo2, we followed its hardware recommendations. As for a human genome (approximately 3 Gb) SOAPdenovo2 required 150Gb, we chose a machine with 104 GB RAM (since the golden mussel's genome is approximately half of human genome's size (1.6 Gb)). This is a n1-highmem-16 machine (with 16 cores), that costs U$0.719/hour to run (considering a hard disk of 1024 GB). If we assume SOAPdenovo2 will take the same amount of time as it took to assemble the human genome (2 days), we will cost us around U$ 34.51 to assemble a genome.

After adjusting the bash file and the machine settings, we proceeded to run the pipeline. However, I got stuck on the creation of config file error: the pipeline is not creating the whole file for the first call of progWriteSoapConfig (despite working just fine for the other calls). We must find out why and fix it.     

#### 09/03/2018
After running the pipeline for the female genome (LF2), we created a new instance (an exact copy of the instance used to assembly LF2's genome) to run the assembly of the male (LF6) genome. After making the necessary adjustment to the _refGuidedDeNovoAssembly_SOAP_withSettings.sh_ file, I left the lab with the process running. 

#### ToDo  
1. Keep track of LF6's assembly process

#### 10/29/2018  
##### Error running SOAPdenovo2
While running de novo assembly for the superblocks, some assemblies got stuck at aio_return error ([as reported by another user here](https://github.com/aquaskyline/SOAPdenovo2/issues/15)). The solution to this problem was to upgrade SOAPdenovo version, since it had been caused by a bug at the older version (that entered an infinite loop when assembling files of size N*32768). The new version used was [this](https://github.com/aquaskyline/SOAPdenovo2). During compilation, we faced [this](https://github.com/aquaskyline/SOAPdenovo2/issues/39) problem, solved as suggested in the linked thread.  

#### 11/06/2018  
##### Status  
Since 10/06/2018, we've been running the alignment of supercontigs to the reference genome in order to remove redundancy.  
##### VM Optimization  
Google Cloud has suggested us to downgrade from a 16 vCPU 104 GB to a 14 vCPU 97 GB instance, since the process being run is underused.     
##### Potential issue  
We are using nucmer version 3 to run the alignment, and its default compilation didn't even allowed us to run the alignment since the reference genome was considered too large. Either way we have recompiled nucmer increasing the upper limit of size for the reference genome (as according to [this](https://sourceforge.net/p/mummer/mailman/message/24384785/) thread) and it is apparently running properly. However, if anything goes wrong, a good option is to use nucmer version 4, which allows multithread and it will probably run even faster. We are not using it since the beggining because the original script from Heidi Lischer was developed using nucmer version 3.  

#### 07/01/2019
##### Status
The nucmer step for the LF6 genome has already been completed, and the same step is still running for the LF2 genome. Since the process has been running for a long time (longer than a month) and since at least three complete contig files generated by SOAP step have already been clustered, we will stop the *mgaps* step and [restart](https://github.com/biobureaubiotech/goldenMusselGender/wiki/Running-Heidi-Lischer's-pipeline#restarting-nucmer-from-mgaps-file) the *nucmer* script from its last step (*postnuc*), that will generate the final *delta* file.  

##### Restarting nucmer from mgaps file  
The LF2 pipeline was stopped at the clustering step, where the *mgaps* program was running and generating a temporary *.mgaps* file. To stop clustering and start the generation of the final *delta* file, we need to:   
1) save the *.maps* file   
2) run the postnuc program to generate the *delta* file:  
    ```console 
    joao_nunes@lf2-genome-assembly: sudo /home/ti_admin/MUMmer3.23/aux_bin/postnuc /home/ti_admin/hlischerLF_10kb.fa Amos_supercontigs.seq Amos_supercontigs < Amos_supercontigs.mgaps
    ``` 
