#### 01/28/2019  
After talking to Marcela about the results we got from the Discovar assembly, she suggested me to give a trial to the software [Cross_genome](https://www.sanger.ac.uk/science/tools/crossgenome), which uses a reference genome from a related species to scaffold. Can we get a better assembly after passing our contigs obtained from Discovar assembly to Cross genome?  

In order to try this hypothesis, we have installed the Cross_genome program at Bioma's computer and executed the following command:  
```console  
bioma@bioma-XPS-8300:~/cross_genome_folder$ /home/bioma/cross_genome_folder/cross_genome -gap 20000 -nodes 20 -score 100 ref_fasta target_fasta merged.fasta
```  

PS: typing the whole path of the cross_genome executable is required to run it without any errors.  

#### 01/30/2019  
##### Running Cross Genome  
Running cross_genome on bioma's pc and right after it finishes run the 1.9 version of samtools to try to get untruncated sort files. (screen session: 19843.cross_genome_lf6). At the same time keeping track of memory usage running the command at the screen session 1214.mem-usage.  

#### 01/31/2019  
After 3 hours running *Cross Genome*, the program has failed with a segmentation fault. A possible reason for the error is lack of memory at the bioma's pc (which has 16GB of RAM). In order to test this hypothesis, we decided to run the *Cross Genome* with the same data at a VM from GCloud (name: lf6-cross-genome, project: Limnoperna fortunei; instance type: n1-standard-8 (8 vCPUs, 30 GB RAM)) using the command:  

```console  ovar's_Assembly
joao_nunes@lf6-cross-genome:~/cross_genome$ { time /home/joao_nunes/cross_genome/cross_genome -gap 20000 -nodes 8 -score 100 ref_fasta target_fasta merged.fasta ; sudo poweroff ; } &> log_cross_genome_lf6_testX.txt  
``` 
The *time* command will record the time taken to run *Cross Genome*, while the poweroff command will turn off the VM instance right after execution.  
However, we got the same segmentation fault error. 

#### 01/31/2019  
##### Running *Cross Genome* at high RAM environment  
In order to test the hypothesis that the segmentation fault was caused by lack of RAM, we tested *Cross Genome* with the same data on a even more powerful machine (name: lf6-cross-genome, project: Limnoperna fortunei; instance type: n1-highmem-16 (16 vCPUs, 104 GB RAM)). 

#### 02/04/2019  
##### Running *Cross Genome* at even higher RAM environment  
Even after using a higher configuration VM instance (n1-highmem-32 (32 vCPUs, 208 GB de memória)), we still got a Segmentation fault error. Since the RAM usage has shown a peak at less than 8 GB (figure below), we ruled out the hypothesis that the error was being caused by a lack of physical memory. Another possible reason for the error is an issue with the input files. I contacted Marcela to check if she already has been successful in running Cross Genome and if Yes, if she could share with me the input files.  

![RAM usage for running Cross Genome](https://github.com/biobureaubiotech/goldenMusselGender/blob/master/images/RAM-usage_kB-per-minute.svg)  

### To Do  
Search about [Align Graph](https://academic.oup.com/bioinformatics/article/30/12/i319/391426), which seems to be a good alternative to Cross Genome.  
  
#### 02/18/2019  
According to Align Graph's paper, it is good practice to do a gap filling step before running Align Graph:
> If scaffolds are inputted, it is usually beneficial to fill them with a gap-filling algorithm before processing them with AlignGraph ([Bao et al., 2014](https://academic.oup.com/bioinformatics/article/30/12/i319/391426#5773866)). 

Then, we installed Gap2Seq using the bioconda installation manager and ran it for the LF2 dataset using the default parameters at the *lf2-genome-assembly-discovar* GCloud VM:  

```console  
joao_nunes@lf2-genome-assembly-discovar:~/Gap2Seq$ Gap2Seq -scaffolds /home/joao_nunes/align_graph/lf2/a.lines.fasta -filled /home/joao_nunes/Gap2Seq/LF2-A_ACAGTG_001.filled.fasta -reads /home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R1_001.fasta,/home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R2_001.fasta -nb-cores 8  
```  

#### 02/20/2019  
#### Switching to another gap-filling algorithm: Sealer  
Gap2Seq was going to take too long to complete according to the execution status. Taking too long to gap-fill large genomes is actually a common problem with gap-filling algorithms, that are usually more suitable for bacterial genomes. A promising alternative algorithm is **Sealer**, part of the Abyss assembly pipeline. According to Sealer description paper:  

> Low sequence coverage, repetitive elements and short read length make de novo genome assembly difficult, often resulting in sequence and/or fragment “gaps” – uncharacterized nucleotide (N) stretches of unknown or estimated lengths. Some of these gaps can be closed by re-processing latent information in the raw reads. Even though there are several tools for closing gaps, they do not easily scale up to processing billion base pair genomes. ([Paulino et al., 2015](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4515008/#MOESM4))

Also according to its paper, Sealer was capable of gap-filling the human genome in 29 hours using a 12 cores, 48G RAM machine. In order to test Sealer for gap-filling our LF2 dataset, we have run the command: 

```console  
joao_nunes@lf2-genome-assembly-discovar:~/abyss-sealer$ abyss-sealer -S /home/joao_nunes/align_graph/lf2/a.lines.fasta -k 80 -o a.lines.gap-filled.fasta -j 16 -b 40G /home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R1_001.fasta /home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R2_001.fasta  
```  

Where k-mer=80bp size was chosen because it has proven to be the optimal size for the human genome, as well as a 40G for the bloom filter. 

The command above has taken only 2h30min to finish. However, it only used one k-mer size to do the gap-filling, instead of the recommended range that goes from average size of the reads down to 40 bp. We then re-run Sealer using as input the output generated from the first round of Sealer (with k-mer=80bp) and using k-mer=40bp this time:  

```console  
joao_nunes@lf2-genome-assembly-discovar:~/abyss-sealer-kmer40$ abyss-sealer -S /home/joao_nunes/abyss-sealer-kmer80/a.lines.gap-filled.fasta_scaffold.fa -k 40 -o a.lines.gap-filled.fasta -j 16 -b 40G /home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R1_001.fasta /home/joao_nunes/align_graph/lf2/LF2-A_ACAGTG_R2_001.fasta  
```  

### To Do
#### How to run Sealer with larger k-mer lengths?  
When we tried to run Sealer with 250, 200 or 150 bp k-mer lengths, we got the following error:    
> abyss-sealer: ../Common/Kmer.h:49: static void Kmer::setLength(unsigned int): Assertion `length <= MAX_KMER' failed.  

The solution to this error seems to be re-configure and recompile Abyss:  
https://www.biostars.org/p/194314/

The compilation is done through: https://github.com/bcgsc/abyss/tree/sealer-release#compiling-abyss-from-github
